notus/scanner/scanner.py

```diff
@@ -16,22 +16,39 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 import logging
-from typing import Generator, Iterable
+from socket import error as SocketError
+from typing import Generator, Iterable, Optional
+
+from paramiko import (
+    AuthenticationException,
+    AutoAddPolicy,
+    BadHostKeyException,
+    SSHClient,
+    SSHException,
+)
+from notus.scanner.models.os.debian import Debian
+from notus.scanner.models.os.distribution import Distribution
+from notus.scanner.models.os.euleros import EulerOS
+from notus.scanner.models.os.suse import SuSE
 
 from notus.scanner.models.packages.deb import DEBPackage
 
 from .errors import AdvisoriesLoadingError
 from .loader import AdvisoriesLoader
 from .messages.message import Message
 from .messages.result import ResultMessage
-from .messages.start import ScanStartMessage
+from .messages.start import ScanHostMessage, ScanStartMessage
 from .messages.status import ScanStatus, ScanStatusMessage
 from .messaging.publisher import Publisher
 from .models.packages.package import Package, PackageAdvisories, PackageType
 from .models.packages.rpm import RPMPackage
 from .models.vulnerability import PackageVulnerability
 
+
+SUPPORTED_DISTRIBUTIONS: list[Distribution] = [Debian, EulerOS, SuSE]
```

Comment:
I think that should be `list[type[Distribution]]` since those are type definition and not initiated distributions.